File: generate_1dca.lua
-- Simulate odd Wolfram rules from W1 to W127
local g = golly()
local n = tonumber(g.getstring("W", "1"))
local rule = "@RULE W"..n.."\n@TABLE\nn_States:3\nneighborhood:Moore\nsymmetries:none\n"
local temp
local s1 = ""
local s2 = ""
for i = 6, 1, -1 do
  temp = i
  for j = 2, 0, -1 do
    if temp >= 2^j then
      temp = temp - 2^j
      s1 = "0"..s1
    else s1 = "1"..s1 end
  end
  temp = i
  for k = 2, 0, -1 do
    if 7 - temp >= 2^k then
      temp = temp + 2^k
      s2 = "2"..s2
    else s2 = "0"..s2 end
  end
  if n > 2^i then
    rule = rule.."0"..s1:sub(2,3).."00000"..s1:sub(1,1).."2\n"
    rule = rule.."0"..s2:sub(2,3).."00000"..s2:sub(1,1).."1\n"
    n = n - 2^i
  end
end
g.setclipstr(rule)

File: generate_1dca.py
## This script generate an ECA rule and emulate it on a torus of width 200. 
## Written by Feng (shouldsee.gem@gmail.com) Feb 2017.
import sys
import golly 

# 193,62,120
head='''@RULE %s
@TABLE
n_states:3
neighborhood:Moore
symmetries:none
var m=0
var p=2
var a={m,1}
var b={m,1}
var c={m,1}
var d={m,1}
var e={m,1}
var f={m,1}
var g={m,1}
var h={m,1}



m,p,p,m,m,m,f,g,p,%s
m,p,1,m,m,m,f,g,p,%s
m,1,p,m,m,m,f,g,p,%s
m,1,1,m,m,m,f,g,p,%s
m,p,p,m,m,m,f,g,1,%s
m,p,1,m,m,m,f,g,1,%s
m,1,p,m,m,m,f,g,1,%s
m,1,1,m,m,m,f,g,1,%s
1,a,b,d,e,f,g,h,c,1
p,a,b,d,e,f,g,h,c,p

''';
rnum=golly.getstring('ECA number','110');
name='W'+rnum;
file=golly.getdir("rules")+name+".rule"

r=bin(int(rnum));
r=r[:1:-1];
r+='0'*(8-len(r));
rule=r;
# rule=[0, 1, 1, 1, 1, 1, 0, 0];

dct=['p','1'];
with open(file,'w') as f:
	f.write(head%((name,)+tuple([dct[int(x)] for x in rule])));
golly.setalgo("RuleLoader")
golly.setrule(name+':T1000,0');
